src/index.ts

```diff
@@ -265,12 +265,21 @@ export class Paginator {
     const callback = parsedArguments.callback!;
     if (parsedArguments.autoPaginate) {
       const results = new Array<{}>();
-      paginator.runAsStream_(parsedArguments, originalMethod)
-          .on('error', callback)
-          .on('data', (data: {}) => results.push(data))
-          .on('end', () => callback(null, results));
+      const promise = new Promise((resolve, reject) => {
+        paginator.runAsStream_(parsedArguments, originalMethod)
+            .on('error', reject)
+            .on('data', (data: {}) => results.push(data))
+            .on('end', () => resolve([results]));
+      });
+      if (callback) {
+        promise.then(
+            results => callback(null, ...results as []),
+            (err: Error) => callback(err));
+      } else {
```

Comment:
This is purely a nitpick, but all the `if/else` action going on here feels unnecessary. Could we simplify the code by doing something like this

```typescript
if (!parsedArguments.autoPaginate) {
  return originalMethod(query, callback);
}

const promise = new Promise((resolve, reject) => {
  // ...
});

if (!callback) {
  return promise;
}

promise.then(/*...*/);
```

Reply:
done, thanks